### ISSUE TITLE
Accessibility: View cart link in notice banner fails color contrast and uses color alone to indicate state

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43615]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The link color for the view cart link in the banner notices fails color contrast and uses only color to differentiate between default and hover states.
<img width="590" alt="Screenshot 2024-01-14 at 5 20 46 PM" src="https://github.com/woocommerce/woocommerce/assets/13153456/271acc18-f98d-4891-bc23-84a68ef6aac9">

### Expected behavior

The link should pass color contrast. Remove or change the percentage for the opacity until it passes color contrast.

On hover, the text-decoration should be set to none so the underline goes away and the hover state is indicated by something other than color.

[Additional information on WCAG Use of Color requirements.](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)

### Actual behavior

See the styles defined for .wc-forward:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss#L46

The color is set to #2f2f2f and there is an opacity of .6. On a light green background, the color before hover is approximately #878B88. This has a [3.37:1 color contrast ratio](https://webaim.org/resources/contrastchecker/?fcolor=878B88&bcolor=F4FFF7), failing WCAG.

On hover, the opacity changes, but there is no other indicator besides color change that the link is being hovered on.

### Steps to reproduce

Review the code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
